src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/View/AbstractViewHandler.cs

```diff
@@ -43,16 +43,28 @@ protected AbstractViewHandler(PropertyMapper mapper)
 
 		public object? NativeView => TypedNativeView;
 
-		public virtual void SetView(IView view)
+
+		public void SetView(IView view)
 		{
 			_ = view ?? throw new ArgumentNullException(nameof(view));
 
+			if (VirtualView != null)
+			{
+				TearDownOldVirtualView(VirtualView);
+				_mapper.UpdateProperty(this, VirtualView, nameof(TearDownOldVirtualView));
+			}
+
 			VirtualView = view as TVirtualView;
-			TypedNativeView ??= CreateView();
+			TypedNativeView ??= CreateNativeView();
 
 			if (!HasSetDefaults)
 			{
-				SetupDefaults();
+				if (TypedNativeView != null)
+				{
+					SetupDefaults(TypedNativeView);
```

Comment:
Changed `SetupDefaults` to take a TypedNativeView so the override doesn't have to null check and it can just operate on the passed in view

Reply:
:+1: